budget-squirrel/src/pages/Login/Login.js

```diff
@@ -17,21 +17,39 @@ export default function Login() {
     const usernameRef = useRef();
     const passwordRef = useRef();
 
+    const [usernameError, setUsernameError] = useState(false);
+    const [passwordError, setPasswordError] = useState(false);
+
     function onLoginClicked() {
         const password = passwordRef.current.value
         const username = usernameRef.current.value
-        authService.login(username, password).then(function() {
+        authService.login(username, password).then(response => {
             history.push(RouteConstants.BUDGET);
+        }).catch(e => {
+            setUsernameError(false);
+            setPasswordError(false);
+            
+            if (e.message === "USER_NOT_FOUND")
```

Comment:
So from a security perspective this probably is a no go. Especially if we are using emails for username. People could farm username's from us. 

For example if they get back `WRONG_PASSWORD` then they know the email is valid. For login error it should be super generic and then maybe include a lock out process for users.

Reply:
Yeah you're right. I was wanting to do that anyways but this is what was in the wireframe. I think just doing a "invalid credentials" is the best way to go.